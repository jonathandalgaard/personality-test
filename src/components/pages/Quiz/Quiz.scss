.quiz-page {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;

  .paper {
    text-align: center;
    width: var(--content-width);
    min-height: 9rem;
  }
}
