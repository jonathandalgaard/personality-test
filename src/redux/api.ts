import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

export const api = createApi({
  reducerPath: 'api',
  baseQuery: fetchBaseQuery({ baseUrl: 'http://teamway.local/api/v1' }),
  endpoints: (builder) => ({
    getQuizzes: builder.query<GetQuizzesApiResponse, void>({
      query: () => '/quizzes',
    }),
    getQuestions: builder.query<GetQuestionsApiResponse, GetQuestionsApiArg>({
      query: ({ quizId }) => `/questions/${quizId}`,
    }),
    getResult: builder.query<GetResultApiResponse, GetResultApiArg>({
      query: ({ score, quizId }) => `/result/${quizId}?score=${score}`,
    }),
  }),
});

export interface MultipleChoiceOption {
  id: number;
  title: string;
  score: string;
}

export interface Question {
  title: string;
  options?: MultipleChoiceOption[];
}

export type GetQuestionsApiResponse = Question[];

export interface GetQuestionsApiArg {
  quizId: number;
}

export interface GetResultApiResponse {
  title: string;
  text: string;
}

export interface GetResultApiArg {
  quizId: number;
  score: number;
}

export interface Quiz {
  id: number;
  title: string;
}

export type GetQuizzesApiResponse = Quiz[];

export const {
  useGetQuizzesQuery,
  useGetQuestionsQuery,
  useGetResultQuery,
} = api;
