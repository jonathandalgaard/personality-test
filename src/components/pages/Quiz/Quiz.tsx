import { useGetQuestionsQuery } from 'redux/api.ts';
import Question from 'components/pages/Quiz/Question/Question';
import { useState } from 'react';

function Quiz() {
  const { data: questions } = useGetQuestionsQuery();
  const [currentQuestion, setCurrentQuestion] = useState<number>(0);
  const [score, setScore] = useState<number>(0);

  const submitAnswer = (score: number) => {
    setScore((prevScore) => prevScore + score);
    setCurrentQuestion((prevQuestion) => prevQuestion + 1);
  }

  if (!questions) {
    return null;
  }

  if (!questions[currentQuestion]) {
    return (
      <div>
        Your score is {score}
      </div>
    );
  }

  return (
    <div>
      <Question question={questions[currentQuestion]} submitAnswer={submitAnswer} />
    </div>
  );
}

export default Quiz;
