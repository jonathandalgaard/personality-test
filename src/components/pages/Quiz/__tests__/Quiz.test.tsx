import renderer from 'react-test-renderer';
import Quiz from 'components/pages/Quiz/Quiz';
import { Provider } from 'react-redux';
import { store } from 'redux/store';
import { BrowserRouter } from 'react-router-dom';

jest.mock('redux/store', () => ({
  __esModule: true,
  store: {
    getState: () => null,
    subscribe: () => null,
    dispatch: () => null,
  },
}));

jest.mock('redux/api', () => ({
  __esModule: true,
  default: () => null,
  useGetQuestionsQuery: () => ({
    data: [
      {
        id: 1,
        title: 'First question',
        options: [
          {
            id: 1,
            title: 'First option',
            score: 1,
          },
          {
            id: 2,
            title: 'Second option',
            score: 4,
          }
        ],
      },
      {
        id: 2,
        title: 'Second question',
        options: [],
      },
    ],
  }),
}));

test('Renders Quiz correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Quiz />
      </BrowserRouter>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
