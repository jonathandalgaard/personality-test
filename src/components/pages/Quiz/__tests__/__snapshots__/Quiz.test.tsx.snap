// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Quiz correctly 1`] = `
<article
  className="quiz-page"
>
  <div
    className="paper"
  >
    <a
      className="back-link"
      href="/"
      onClick={[Function]}
    >
      <svg
        className="lucide lucide-corner-up-left"
        fill="none"
        height="1.125em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2.5}
        viewBox="0 0 24 24"
        width="1.125em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 14 4 9 9 4"
        />
        <path
          d="M20 20v-7a4 4 0 0 0-4-4H4"
        />
      </svg>
      Go back
    </a>
    <div
      className="question"
    >
      <h2>
        First question
      </h2>
      <ul
        className="options"
      >
        <li>
          <button
            className="button"
            onClick={[Function]}
          >
            <span
              className="option-index"
            >
              1
              .
            </span>
             
            First option
          </button>
        </li>
        <li>
          <button
            className="button"
            onClick={[Function]}
          >
            <span
              className="option-index"
            >
              2
              .
            </span>
             
            Second option
          </button>
        </li>
      </ul>
    </div>
  </div>
</article>
`;
